### PR TITLE
[KOGITO-1031] Add Maven local deploy

### PR DESCRIPTION
**Pull request by Kevin-Mok**
See: https://issues.redhat.com/browse/KOGITO-1031

This is an accompanying PR to [the one opened in kogito-pipelines](https://github.com/kiegroup/kogito-pipelines/pull/46). 

## Changes
- deploys all artifacts locally before zipping them altogether to push to Nexus
  * added to provide authentication to push artifacts to PR artifact repository
- replace hardcoded "kogito-examples" string with environment variable

## Testing
I was able to [deploy the artifacts](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/kmok/job/KOGITO-1031-full/job/kogito-examples-deploy/26/execution/node/170/log/?consoleFull) to the PR test repository using the code from this PR.

## Requirements
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -144,7 +147,7 @@ pipeline {
         stage('Check kogito-examples with events') {
             steps {
                 script {
-                    sh 'cp -r kogito-examples kogito-examples-events'
+                    sh "cp -r ${getRepoName()} ${getRepoName()}-events"
                     getMavenCommand()
```
Comment (radtriste) [inline Jenkinsfile.deploy]: I guess this is a bug coming from previous code ...
```suggestion
                    getMavenCommand("${getRepoName()}-events")
```

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -160,7 +163,12 @@ pipeline {
         stage('Deploy artifacts') {
             steps {
                 script {
-                    if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
+                    if (params.MAVEN_DEPLOY_REPOSITORY && params.MAVEN_REPO_CREDS_ID) {
+                        // Deploy to specific repository with credentials
+                        runMavenDeployLocally(getMavenCommand())
```
Comment (radtriste) [inline Jenkinsfile.deploy]: I guess for clarity in the code, we should put the `directory` parameter as mandatory ...

```suggestion
                        runMavenDeployLocally(getMavenCommand(getRepoName()))
```

File: Jenkinsfile.deploy [hunk 3]
```diff
@@ -160,7 +163,12 @@ pipeline {
...
+                        maven.uploadLocalArtifacts(params.MAVEN_REPO_CREDS_ID, getLocalDeploymentFolder(), getMavenRepoZipUrl())
+                    } else if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
+                        // Normal deploy
                         runMavenDeploy(getMavenCommand())
```
Comment (radtriste) [inline Jenkinsfile.deploy]: I guess for clarity in the code, we should put the `directory` parameter as mandatory ...

```suggestion
                        runMavenDeployLocally(getMavenCommand(getRepoName()))
```

File: Jenkinsfile.deploy [hunk 4]
```diff
@@ -286,11 +298,14 @@ void setDeployPropertyIfNeeded(String key, def value){
     }
 }
 
-MavenCommand getMavenCommand(String directory = 'kogito-examples'){
+MavenCommand getMavenCommand(String directory = ''){
+    if(!directory) {
+        directory = getRepoName()
+    }
```
Comment (radtriste) [inline Jenkinsfile.deploy]: I guess for clarity in the code, we should put the `directory` parameter as mandatory ...
```suggestion
MavenCommand getMavenCommand(String directory){
```

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
Jenkins retest this

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**